### DISCUSSION TITLE
RBE-weighted doses for Radiosensitizers

### DISCUSSION THREAD

**Discussion by naninano1** [e0404/matRad#546] [Q&A]
Hi,

I was wondering how we can apply a certain RBE value, caused by a radiosensitizer agent, into the CTV (in voxel or whole CTV scale) in dose calculations to see RBE-weighted dose distribution and DVHs for a PHOTON beam.

**Comment by naninano1** [comment]
Hi @wahln ,
I hope you don't mind mentioning you here. I was wondering if you could please guide me in this regard.
Thank you.

**Comment by wahln** [comment]
Hi @nanino1,
there is no pre-defined way to do it in matRad, I would just work on the output cubes you get after optimization (if you want to optimize for RBE-weighted photon dose, things would get a little messier, but it would also be possible). Steps would be:

1. Do a normal photon optimization, you obtain a `resultGUI` struct that has the cube `physicalDose` stored. 
2. Get the index list of your ct voxels (let's call it `ix` and the corresponding vector of `RBE` for each index
3. Create the RBE map from the command line: `resultGUI.RBE = ones(size(resultGUI.physicalDose)); resultGUI.RBE(ix) = RBE;` This makes sure that the RBE is 1 everywhere you have not assigned a special value.
4. Create the RBE weighted dose: `resultGUI.RBExD = resultGUI.physicalDose .* resultGUI.RBE(ix) = RBE;`

After this, you should be able to select your RBE-weighted dose cube in the matRadGUI after refreshing.

**Reply by naninano1** [reply]
Hi Niklas,
Thank you very much for your reply.
For part 2, I did `ix = cst{1,4}{1}; RBE= [1.2];` for the first index , GTV.
However, when I run `resultGUI.RBExD = resultGUI.physicalDose .* resultGUI.RBE(ix) = RBE;`, I get this error:
"Error: Incorrect use of '=' operator. To assign a value to a variable, use '='. To compare values for equality, use '=='."

**Reply by wahln** [reply]
Sorry, I was not really available last week. I see you could solve the issue?

**Reply by naninano1** [reply]
Hi Niklas,
No problem. Thanks for your consideration. 
Yes, I could solve the issue. I redefined ix and RBE, also the dose map
